internal/fullnode/pvc_builder_test.go

```diff
@@ -148,30 +149,81 @@ func TestBuildPVCs(t *testing.T) {
 		}
 	})
 
-	t.Run("pvc auto scale", func(t *testing.T) {
-		for _, tt := range []struct {
-			SpecQuant, AutoScaleQuant, WantQuant string
-		}{
-			{"100G", "99G", "100G"},
-			{"100G", "101G", "101G"},
-		} {
-			crd := defaultCRD()
-			crd.Spec.Replicas = 1
-			crd.Spec.VolumeClaimTemplate = cosmosv1.PersistentVolumeClaimSpec{
-				Resources: corev1.ResourceRequirements{
-					Requests: corev1.ResourceList{corev1.ResourceStorage: resource.MustParse(tt.SpecQuant)},
-				},
-			}
-			crd.Status.SelfHealing.PVCAutoScale = &cosmosv1.PVCAutoScaleStatus{
-				RequestedSize: resource.MustParse(tt.AutoScaleQuant),
+	t.Run("pvc auto scale with padding", func(t *testing.T) {
+		t.Run("given auto scale size less then current size", func(t *testing.T) {
+			for _, tt := range []struct {
+				SpecQuant, AutoScaleQuant, WantQuant string
+			}{
+				{"100G", "97G", "100G"},
+			} {
+				crd := defaultCRD()
+				crd.Spec.Replicas = 1
+				crd.Spec.VolumeClaimTemplate = cosmosv1.PersistentVolumeClaimSpec{
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{corev1.ResourceStorage: resource.MustParse(tt.SpecQuant)},
+					},
+				}
+				crd.Status.SelfHealing.PVCAutoScale = &cosmosv1.PVCAutoScaleStatus{
+					RequestedSize: resource.MustParse(tt.AutoScaleQuant),
+				}
+
+				pvcs := BuildPVCs(&crd)
+				require.Len(t, pvcs, 1, tt)
+
+				want := corev1.ResourceList{corev1.ResourceStorage: resource.MustParse(tt.WantQuant)}
+				require.Equal(t, want.Storage().Value(), pvcs[0].Object().Spec.Resources.Requests.Storage().Value(), tt)
 			}
```

Comment:
because of the Int64, Dec values we need to compare Value() directly because some fields do not get updated after the calculations but I assume k8s uses Value() anyway.